python/ray/serve/api.py

```diff
@@ -1046,163 +1049,163 @@ def decorator(cls):
     return decorator
 
 
-class ServeDeployment(ABC):
-    @classmethod
-    def deploy(cls, *init_args) -> None:
+class ServeDeployment:
```

Comment:
I needed to make this a normal class in order to define `__str__` and `__cmp__` methods. The other way would be to define a metaclass but cloudpickle really didn't like me doing that so I reverted to this.

Reply:
Sounds good.  Initially, why was it better to make it return a class instead of a class instance?  In other words, what's the downside of this change?

Reply:
Eh there isn't a huge difference (everything in python is an object anyways :)), but it is more intuitive to return a class rather than an object from the decorator given that an object was never explicitly constructed.

Reply:
I guess we needed it to test equality of the deployment list dict.

Reply:
Yep exactly! The `__str__` is just an additional niceity